funcx_endpoint/funcx_endpoint/executors/high_throughput/interchange.py

```diff
@@ -572,10 +578,14 @@ def _command_server(self, kill_event):
                 command = Message.unpack(buffer)
                 if command.type not in COMMAND_TYPES:
                     logger.error("Received incorrect message type on command channel")
-                    self.command_channel.send(bytes())
-                    continue
+                    reply = BadCommand(f"Unknown command type: {command.type}")
+
+                elif command.type is MessageType.TASK_CANCEL:
+                    logger.info(f"[COMMAND] Received TASK_CANCEL for task_id:{command.task_id}")
+                    self.enqueue_task_cancel(command.task_id)
+                    reply = command
 
-                if command.type is MessageType.HEARTBEAT_REQ:
+                elif command.type is MessageType.HEARTBEAT_REQ:
                     logger.info("[COMMAND] Received synchonous HEARTBEAT_REQ from hub")
                     logger.info(f"[COMMAND] Replying with Heartbeat({self.endpoint_id})")
                     reply = Heartbeat(self.endpoint_id)
```

Comment:
separate from task cancel - could be nice to have a final `else` on this

Reply:
Fixed.